lib/wickedbase.pm

```diff
@@ -489,17 +489,34 @@ sub setup_team {
     $self->get_from_data('wicked/ifcfg/ifcfg-eth0-hotplug',     $cfg_ifc0);
     $self->get_from_data('wicked/ifcfg/ifcfg-eth0-hotplug',     $cfg_ifc1);
     $self->get_from_data('wicked/teaming/ifcfg-team0-' . $mode, $cfg_team0);
-    file_content_replace($cfg_team0, ipaddr4 => $self->get_ip(type => 'host', netmask => 1), ipaddr6 => $self->get_ip(type => 'host6', netmask => 1), iface0 => $iface0, iface1 => $iface1);
+    my $ipaddr4  = $self->get_ip(type => 'host',  netmask       => 1);
+    my $ipaddr6  = $self->get_ip(type => 'host6', netmask       => 1);
+    my $ping_ip4 = $self->get_ip(type => 'host',  is_wicked_ref => 1);
+    my $ping_ip6 = $self->get_ip(type => 'host6', is_wicked_ref => 1);
+    file_content_replace($cfg_team0, ipaddr4 => $ipaddr4, ipaddr6 => $ipaddr6, iface0 => $iface0, iface1 => $iface1, ping_ip4 => $ping_ip4, ping_ip6 => $ping_ip6);
 
     $self->wicked_command('ifup', 'all');
 }
 
+sub get_team_active_link {
+    my ($self, $team_link) = @_;
+    my $out = script_output("teamdctl $team_link state view");
+    record_info('teamdctl', $out);
+    my @lines = split(/\n/, $out);
+    return $1 if ( $out =~ m/active port: (\w+)/);
+    return;
+}
+
 sub validate_interfaces {
-    my ($self, $interface, $iface0, $iface1) = @_;
-    die("Missing interface $interface") unless ifc_exists($interface);
+    my ($self, $interface, $iface0, $iface1, $ping) = @_;
+    $ping //= 1;
+    if (!$self->ifc_is_up($interface)) {
```

Comment:
so now I need to do extra hop to network_utils.pm to see what exactly we are doing to check that interface is up :( . how this could be called "increase readability" when now I need **more** time to understand what happening ?

Reply:
hm, for me, if a function is called ifc_is_up() I don't care how it check the upstate, but trust that the interface is up...

Reply:
I personally like it better like this, if I use ifc_exists, you still need to go to network_utils to see what it does. The name is quite explicit anyways and you can get the idea what the function does by its name